lib/plugins/villager.js

```diff
@@ -150,12 +170,22 @@ function inject (bot, { version }) {
     }
   }
 
-  function putRequirements (window, Trade, cb) {
-    deposit(window, Trade.inputItem1.type, Trade.inputItem1.metadata, Trade.inputItem1.count, 0, (err) => {
+  function putRequirements (window, Trade, count, cb) {
+    let input1 = 0
+    if (!window.slots[0] || window.slots[0].count < Trade.inputItem1.count) {
+      input1 = Math.min(Trade.inputItem1.stackSize, Trade.inputItem1.count * count)
+      if (window.slots[0]) input1 -= window.slots[0].count
+    }
```

Comment:
I think that this can be simplified and it for sure can look better

Reply:
Is this what you mean?
```js
    let input1 = 0
    if (!window.slots[0]) {
      input1 = Math.min(Trade.inputItem1.stackSize, Trade.inputItem1.count * count)
    } else if (window.slots[0].count < Trade.inputItem1.count) {
      input1 = Math.min(Trade.inputItem1.stackSize, Trade.inputItem1.count * count) - window.slots[0].count
    }
```

Reply:
Yeah, it's a bit more readable, you could even do like that or something similar. What I mean is that there's just too much similar long variables that makes this part of code hard to read
```js
const [slot] = window.slots
const { count: tradeCount1, stackSize: stackSize1 } = Trade.inputItem1
const realCount1 = Math.min(stackSize1, tradeCount1 * count)

const input1 = !slot 
  ? realCount1
  : (slot.count < tradeCount1 ? realCount1 - slot.count : 0)
```

Reply:
Yeah that's much clear